### ISSUE TITLE
Are Pragma Dimension and overlap Mapping really necessary at compilation time?

### ISSUE THREAD

**Issue by SergioMartin86** [SergioMartin86/bamboo6#37]
Scott asked:

"That's a run time issue. Don't you need to know the # of dimensions at compile time to handle, say,
nearest neighbor connections?
What else is dimension really used for ?"

**Comment by SergioMartin86** [comment]
As far as I can see, both the mapping (nearest neighbor/alltoall/etc) and the dimension number can be set at runtime. This may take some tweaking of both Bamboo and Tarragon, but it's completely plausible, and can simplify the user's responsibilities and the translated code.

**Comment by SergioMartin86** [comment]
No, this is possible. However, since templated functions were - rather unnecessarily - used in many class definitions Tarragon, it is going to be pretty tricky to replace them by constructor parameters. It is going to take a while, but it is doable.

**Comment by scottb5612** [comment]
OK, let's keep this issue open, for later discussions.

**Comment by SergioMartin86** [comment]
I've just made this change. Now both dimension and layout options are passed as runtime arguments.

This is a big one, and there's a lot to explain. I think it will be best if we can talk this over the phone. In the meantime, see that all examples have been adapted to the new method. Also note that the main() method has been drastically reduced :package: 

There's only a strange thing happening with the struct_type example. Looks like a simple bug, but I'm not closing this issue until I solve it.

**Comment by SergioMartin86** [comment]
aah, it was just a configuration mistake. This is done.

**Comment by scottb5612** [comment]
OK, let's talk about this. In the meantime:
I can build and run the 7 point app on stampede and sorken, but not at NYU
Do you have a script that runs all the tests, and checks if they pass?

*...  both dimension and layout options are passed as runtime arguments.
This is a big one, and there's a lot to explain. I think it will be best if we can talk this over the phone. *

**Comment by SergioMartin86** [comment]
There's not a testing script yet because the testcases have not yet been formalized. I want to do this (along the website, configuration/user manuals, distributable source) as soon as we get Cart3D working
